### PR TITLE
Infra: install g++11. Upgrade tromp.

### PR DESCRIPTION
**Pull request by snellers**

### CODE REVIEW

File: infra/terraform/xania.tf [hunk 1]
```diff
@@ -187,7 +187,7 @@ resource "aws_iam_role_policy_attachment" "xania_attach_policy" {
 }
 
 resource "aws_instance" "XaniaNode" {
-  ami = "ami-03bd6c18a631bc24a"
+  ami = "ami-04505e74c0741db8d"
```
Comment (snellers) [inline infra/terraform/xania.tf:190]: Just bringing it up to date. The amd 20.04 image in us-east-1,  https://cloud-images.ubuntu.com/locator/ec2/

File: conanfile.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@
 catch2/2.13.7
 fmt/8.0.1
 ms-gsl/3.1.0
-trompeloeil/41
+trompeloeil/42
```
Comment (snellers) [inline conanfile.txt:5]: rollbear has a few fixes & improvements

Comment (mattgodbolt) [reply conanfile.txt:5]: 👍🏻

File: infra/terraform/xania.tf (hunk 1 above)
Comment (mattgodbolt) [reply infra/terraform/xania.tf:190]: will applying this change recreate the AMI though? (we'll have to ensure everything's backed up and restored if so)

**Review by mattgodbolt — APPROVED** [review]
one comment about the tf change

File: infra/terraform/xania.tf (hunk 1 above)
Comment (snellers) [reply infra/terraform/xania.tf:190]: I'll undo it -- for the moment, we actually want the option to be able to make other infra changes (e.g. networking) by rerunning terraform, and without forcing a change to the  base image.

**Pull request merged**